neps/optimizers/models/deepGP.py

```diff
@@ -0,0 +1,572 @@
+        )
+
+    def encode_configs(
+        self,
+        configs: list[SearchSpace],
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        x_buffer = torch.empty(
+            (len(configs), self.output_dim),
+            device=self.device,
+            dtype=torch.float32,
+        )
+
+        # Normals are just fill the columns with the normalized values
+        for i, (hp_name, hp) in enumerate(self.numericals.items()):
+            budget_tensor = torch.tensor(
+                [config[hp_name].value for config in configs],
+                device=self.device,
+                dtype=torch.float32,
+            )
+
+            x_buffer[:, i] = (budget_tensor - hp.lower) / (hp.upper - hp.lower)
+
+        # Categoricals is a bit harder, we create a tensor with all the indices (values)
+        # as we did above, but then we sub-select the portion of the buffer for that categorical
+        # before inserting the one-hot encoding.
+        offset = len(self.numericals)
+        for hp_name, hp in self.categoricals.items():
+            budget_tensor = torch.tensor(
+                [config[hp_name]._value_index for config in configs],  # type: ignore
+                device=self.device,
+                dtype=torch.float64,
+            )
+
+            # .. and insert one-hot encoding (ChatGPT solution, verified locally)
+            portion = x_buffer[:, offset : offset + len(hp.choices)]
+            portion.scatter_(1, budget_tensor.unsqueeze(1), 1)
+
+            offset += len(hp.choices)
+
+        # Finally, ... budgets
+        budgets = [config.fidelity.value for config in configs]  # type: ignore
+        budget_tensor = torch.tensor(budgets, device=self.device, dtype=torch.float32)
+        if self.fidelity_bounds:
+            assert self.space.fidelity is not None
+            _min = self.space.fidelity.lower
+            _max = self.space.fidelity.upper
+            budget_tensor.sub_(_min).div_(_max - _min)
+
+        return x_buffer, budget_tensor
+
+    def encode_learning_curves(self, learning_curves: list[list[float]]) -> torch.Tensor:
+        lc_height = len(learning_curves)
+        lc_width = max(
+            max(len(lc) for lc in learning_curves), self.min_learning_curve_length
+        )
+        lc_buffer = torch.full(
+            (lc_width, lc_height),
+            self.learning_curve_pad_value,
+            device=self.device,
+            dtype=torch.float32,
+        )
+
+        for i, lc in enumerate(learning_curves):
+            lc_buffer[: len(lc), i] = torch.tensor(
+                lc, device=self.device, dtype=torch.float32
+            )
+
+        return lc_buffer
+
+    def encode_y(
+        self, y: list[float]
+    ) -> tuple[torch.Tensor, None | tuple[int | float, int | float]]:
+        tensor = torch.tensor(y, device=self.device, dtype=torch.float32)
+        if self.fidelity_bounds:
+            _min, _max = tensor.min(), tensor.max()
+            tensor.sub_(_min).div_(_max - _min)
```

Comment:
I think we should drop output normalization, I've added it as an option early on, but we never used it. It also makes the problem harder since we usually expect `y` to be on the same scale as the learning curve